Source/UnitTests/Core/PatchAllowlistTest.cpp

```diff
@@ -69,12 +77,23 @@ TEST(PatchAllowlist, VerifyHashes)
     std::string game_id = file.virtualName.substr(0, file.virtualName.find_first_of('.'));
     std::vector<PatchEngine::Patch> patches;
     PatchEngine::LoadPatchSection("OnFrame", &patches, ini_file, Common::IniFile());
+    std::vector<Gecko::GeckoCode> geckos = Gecko::LoadCodes(Common::IniFile(), ini_file);
+    std::vector<ActionReplay::ARCode> action_replays =
+        ActionReplay::LoadCodes(Common::IniFile(), ini_file);
     // Filter patches for RetroAchievements approved
     ReadEnabledOrDisabled<PatchEngine::Patch>(ini_file, "OnFrame", false, &patches);
     ReadEnabledOrDisabled<PatchEngine::Patch>(ini_file, "Patches_RetroAchievements_Verified", true,
                                               &patches);
+    ReadEnabledOrDisabled<Gecko::GeckoCode>(ini_file, "Gecko", false, &geckos);
+    ReadEnabledOrDisabled<Gecko::GeckoCode>(ini_file, "Gecko_Enabled", false, &geckos);
+    ReadEnabledOrDisabled<Gecko::GeckoCode>(ini_file, "Gecko_RetroAchievements_Verified", true,
+                                            &geckos);
```

Comment:
```suggestion
    ReadEnabledOrDisabled<Gecko::GeckoCode>(ini_file, "Gecko", false, &geckos);
    ReadEnabledOrDisabled<Gecko::GeckoCode>(ini_file, "Gecko_RetroAchievements_Verified", true,
                                            &geckos);
```

Whether the user has chosen to enable a specific code should have no impact on whether it should be verified.

Reply:
I'll have to look through the codes and find them again but I had at least one code that was in a Gecko_Enabled block but not a Gecko block by default. I suppose I should check Gecko_Disabled too then?

Reply:
For context for others, here's what you wrote on Discord:

> okay so I dug into this https://github.com/dolphin-emu/dolphin/pull/12955#discussion_r1862561883 to remind myself and it's the Nintendo Channel issue. All three inis for Nintendo Channel do this:
> 
> ```
> [Gecko]
> $SSL Patch [Palapeli]
> ...
> 
> [Gecko_Enabled]
> $SSL Patch
> ```
> 
> The result is that when the unit test does the following:
> 
> ```
>     std::vector<Gecko::GeckoCode> geckos = Gecko::LoadCodes(Common::IniFile(), ini_file);
> ...
>     ReadEnabledOrDisabled<Gecko::GeckoCode>(ini_file, "Gecko", false, &geckos);
>     ReadEnabledOrDisabled<Gecko::GeckoCode>(ini_file, "Gecko_RetroAchievements_Verified", true,
>                                             &geckos);
> ```
> 
> it loads separate codes for `SSL Patch [Palapeli]` and `SSL Patch`, disables `SSL Patch [Palapeli]`, and then assumes `SSL Patch` is `Gecko_RetroAchievements_Verified` because nothing ever disables it when the test starts checking patches in the allowlist.

I'm not sure how both the `SSL Patch [Palapeli]` and `SSL Patch` variants are ending up in the list, but either way: As I understand it, what you're trying to do is to set `enabled` to false for every code because you want to reuse the `enabled` variable for a different purpose. There's a simpler way to do this: Instead of using `ReadEnabledOrDisabled` to set `enabled` to false, you can loop through the codes and set `enabled` to false for all of them. This will be less fragile.